sdk/servicebus/Azure.Messaging.ServiceBus/src/Addons/ServiceBusMessageExtensions.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Azure.Messaging.ServiceBus.Addons
+{
+    /// <summary>
+    /// Extension methods for <see cref="ServiceBusMessage"/>.
+    /// </summary>
+    public static class ServiceBusMessageExtensions
+    {
+        /// <summary>
+        /// Gets the Data body type from an AMQP <see cref="ServiceBusMessage"/>.
+        /// </summary>
+        /// <param name="message">The message to use.</param>
+        /// <returns></returns>
+        public static ReadOnlyMemory<byte> GetAmqpDataBody(this ServiceBusMessage message) =>
+            message.AmqpData;
```

Comment:
What happens if we have a similar concept for other protocols?   With this pattern, we'd have to add new properties for each protocol-specific type.   I think there's merit to having the `TransportBody` or `TransportProperties` or similar abstraction tied to this.

The `Azure.Messaging.ServiceBus.Transports.Amqp` namespace could have the extensions and the concrete `AmqpTransportBody` so there would be a reasonable level of visibility between them.

Reply:
I will do this. The reason I didn't do it at first was because it creates another instance type (DTO) to simply hold these 3 properties. So, it is an overhead that is added for every message compared to what we have right now. Again, I'm not used to thinking about things this way, but I'm learning. Does it make any realistic difference to have extra instances like this one?

Reply:
I created `AmqpTransportBody`. This implemented `ITransportBody`. A `ServiceBusMessage` contains an internal `TransportBody`. The `Body` property on `ServiceBusMessage` simply exposes the `Body` property of the `TransportBody` via the `ITransportBody` interface.

Reply:
I wonder if it would be more consistent to use an abstract base class the way we do for all of the TransportSender/TransportReceiver stuff.

Reply:
This was changed to an abstract base class.